### PR TITLE
shock mine improvements

### PR DESCRIPTION
**Pull request by Slava0135**
Now shock mine won't trigger if it's on low hp (less than 1 charge). They will blink if so.
Peek 2020-11-30 18-18

### CODE REVIEW

**Comment by Slava0135** [comment]
*not my fault

File: core/src/mindustry/world/blocks/defense/ShockMine.java [hunk 1]
```diff
@@ -37,20 +40,30 @@ public void drawTeam(){
...
+            if(isActive() || progress < 20f){
+                Draw.color(team.color);
+                Draw.alpha(0.5f);
+                Fill.rect(x, y, 2f, 2f);
+                Draw.color();
+            }
+            if(!isActive()){
+                progress += delta();
+                progress %= blinkDelay;
+            }
```
Comment (summetdev) [inline core/src/mindustry/world/blocks/defense/ShockMine.java:52]: it's better to use `else` here.
```suggestion
            }else{
                progress += delta();
                progress %= blinkDelay;
            }
```

File: core/src/mindustry/world/blocks/defense/ShockMine.java [hunk 2]
```diff
@@ -29,6 +29,9 @@ public ShockMine(String name){
 
     public class ShockMineBuild extends Building{
```
Comment (summetdev) [inline core/src/mindustry/world/blocks/defense/ShockMine.java:31]: empty line in top of class.
```suggestion
```

File: core/src/mindustry/world/blocks/defense/ShockMine.java [hunk 3]
```diff
@@ -37,20 +40,30 @@ public void drawTeam(){
...
+            if(!isActive()){
+                progress += delta();
+                progress %= blinkDelay;
+            }
         }
 
         @Override
         public void unitOn(Unit unit){
-            if(enabled && unit.team != team && timer(timerDamage, cooldown)){
+            if(enabled && unit.team != team && timer(timerDamage, cooldown) && isActive()){
```
Comment (summetdev) [inline core/src/mindustry/world/blocks/defense/ShockMine.java:57]: maybe put `isActive()` after `enabled`? so it doesn't need to execute more height statements.

**Review by summetdev — CHANGES_REQUESTED** [review]

File: core/src/mindustry/world/blocks/defense/ShockMine.java (hunk 1 above)
Comment (Slava0135) [reply core/src/mindustry/world/blocks/defense/ShockMine.java:51]: this will break stuff

**Comment by DeltaNedas** [comment]
good

**Comment by joshuaptfan** [comment]
The square light is ugly, how about a smaller circle light, same size as the 4 holes? And it should be red, no need for team color as it will already have the team marker when it belongs to another team.

**Comment by sample-text-here** [comment]
[I don't think shock mines are supposed to show their team color](https://github.com/Anuken/Mindustry/blob/d64f603e08ea27804bdc25026392ad30b1470563/core/src/mindustry/world/blocks/defense/ShockMine.java#L33-L35)

**Comment by Bohaska** [comment]
@Slava0135 update this pr

**Review by sk7725 — CHANGES_REQUESTED** [review]
1. Why not only render the blink by `if Time.time % certainamount > certainamount / 2`?
2. Shock mines are supposed to be hard to spot. Make the blink only draw for mines on your team, and keep the led "off" by default, so it is hard to tell your shock mines from the enemy's(which is why team indicators are nonexistant on this block)
3. Why not use a textureregion instead of a solid square, for mods and aesthetics?

**Comment by Slava0135** [comment]
> 1. Why not only render the blink by `if Time.time % certainamount > certainamount / 2`?
> 2. Shock mines are supposed to be hard to spot. Make the blink only draw for mines on your team, and keep the led "off" by default, so it is hard to tell your shock mines from the enemy's(which is why team indicators are nonexistant on this block)
> 3. Why not use a textureregion instead of a solid square, for mods and aesthetics?

2 is Pretty reasonable. I will move the blink effect to drawTeam. Idk about 3. People could just override draw methods if they want. And 1 - there is 2 divisions, while

**Comment by Slava0135** [comment]
Oh I forgot what drawTeam does. It is called on block if you are on enemy team not ally.

**Comment by Slava0135** [comment]
image
idk if it's bright enough to be noticable at close look but not too obvious

**Comment by Slava0135** [comment]
Peek 2020-12-09 20-16

Is there a way to remove "breaking" layer?

**Comment by Slava0135** [comment]
Peek 2020-12-16 14-18
no cracks for now

**Comment by Slava0135** [comment]
*perhaps I should just copy unit blinking health bar thing

**Pull request closed without merging**